### ISSUE TITLE
Request is invalid - NextToken must not be empty

### ISSUE THREAD

**Issue by mtarique** [jlevers/selling-partner-api#607]
## Error:[400] { "errors": [ { "code": "InvalidInput", "message": "Request is invalid - NextToken must not be empty", "details": "" } ] }

```
# Your error here. PLEASE make sure to redact secrets from your error log!
```

## Code

```php
$api = new SellingPartnerApi\Api\FbaInboundV0Api($config);

$resNextToken = $api->getShipments('NEXT_TOKEN', ['NextToken' => $nextToken]);
```

## Seller Central SP API config page screenshot

*your screenshot here*

**Comment by DomuITteam** [comment]
Looking at the documentation, you still need **query_type** and **marketplace_id** in your request, as these are non-optional. So your code might need to look like this:

```
$resNextToken = $api->getShipments($query_type, $marketplace_id, $shipment_status_list, $shipment_id_list, $last_updated_after, $last_updated_before, $next_token);
```

If you don't need to pass other optional parameters, then:

```
$resNextToken = $api->getShipments($query_type, $marketplace_id, '', '', '', '', $next_token);
```

**Comment by mtarique** [comment]
Hi @DomuITteam,

Thanks for your response.

But when I am including these params it returns same shipments twice

We don't have next pages still receiving next token which contains empty next page.

**Comment by DomuITteam** [comment]
To start with, check if the payload from **getShipments** contains **NextToken** value, and if so, if it's empty or not. 

If both are true, try this instead:

```
$next_token = $payload->NextToken;
$resNextToken = $api->getShipments(NEXT_TOKEN, $marketplace_id, '', '', '', '', $next_token);
```

**Comment by mtarique** [comment]
Yes payload contains next token and its not empty but the issue is when I am making next request with next token it returns all the results similar to original one. Which makes the final list duplicate for each shipment.

**Comment by avantcoretech** [comment]
> Yes payload contains next token and its not empty but the issue is when I am making next request with next token it returns all the results similar to original one. Which makes the final list duplicate for each shipment.

Were you able to solve this problem? I am getting the same problem with next token. getting the same data with the next request using next token

**Comment by MuhammadDanyal** [comment]
> Yes payload contains next token and its not empty but the issue is when I am making next request with next token it returns all the results similar to original one. Which makes the final list duplicate for each shipment.

Were you able to solve this problem? I am getting the same problem with next token. getting the same data with the next request using next token

**Comment by iajrz** [comment]
This issue has been inactive for a while; since the 6.x version has been wholly superseded by the 7.x branch, I'm closing this issue.
